presto-mysql/src/main/java/io/prestosql/plugin/mysql/MySqlClient.java

```diff
@@ -268,6 +268,21 @@ else if (varcharType.getBoundedLength() <= 16777215) {
         return super.toWriteMapping(session, type);
     }
 
+    private boolean isGTIDMode(Connection connection)
+    {
+        try (java.sql.Statement statement = connection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery("show variables like 'gtid_mode'");
+            if (resultSet.next()) {
+                return resultSet.getString("Value").equalsIgnoreCase("ON");
```

Comment:
`ON_PERMISSIVE` is also possible value.

Reply:
I use a `not OFF mode` check here to consider `OFF_PERMISSIVE `、`ON__PERMISSIVE` and `ON` modes at the same time. Mysql may use GTID-based replication in `OFF_PERMISSIVE` mode according to [MySQL dev doc](https://dev.mysql.com/doc/refman/5.7/en/replication-options-gtids.html)